Source/EasyNetQ/DefaultRpc.cs

```diff
@@ -147,24 +147,22 @@ protected void RegisterResponseActions<TResponse>(string correlationId, TaskComp
         var responseAction = new ResponseAction(
             message =>
             {
-                var msg = (IMessage<TResponse>)message;
+                var messageOfTResponse = (IMessage<TResponse>)message;
 
                 var isFaulted = false;
                 var exceptionMessage = "The exception message has not been specified.";
-                if (msg.Properties.HeadersPresent)
+                if (messageOfTResponse.Properties is { HeadersPresent: true, Headers: { } })
```

Comment:
```suggestion
                if (messageOfTResponse.Properties is { HeadersPresent: true })
```

Reply:
It will not compile because of a dereference of potential null value.

Reply:
Then yes, but in addition to that I prefer removing *Present properties, because they are useless now.

Reply:
ok